TMBuilder/src/main/java/com/ebay/tool/thinmodelgen/gui/menu/export/DeveloperMockExport.java

```diff
@@ -0,0 +1,619 @@
+            // $.root.next[1].step[*].key - will set/overwrite every step.key on the first next index.
+            populateJsonMapWithMockValues(validationSetModel);
+
+            // 4) Write the JSON to file.
+            String fileName = String.format(DEVELOPER_MOCK_FILE_NAME_FORMAT, validationSetModel.getValidationSetName());
+            JSONObject jsonObject = new JSONObject(jsonMap);
+            String json = jsonObject.toString();
+            // TODO: write jsonMap, as JSON, to file.
+        }
+    }
+
+    /**
+     * For testing.
+     * @return Unmodifiable map.
+     */
+    protected Map<String, Object> getJsonMap() {
+        return Collections.unmodifiableMap(jsonMap);
+    }
+
+    /**
+     * For testing - access to map. Cannot modify source map with copy returned.
+     * @return Copy of map.
+     */
+    protected TreeMap<String,Integer> getArrayPathToArraySizeMap() {
+        return new TreeMap<>(arrayPathToArraySizeMap);
+    }
+
+    /**
+     * Populate the JSON map, that will be serialized as the JSON mock, with the parsed array data. This operation will
+     * initialize each of the array nodes to their expected sizes. Expected that populateArrayPathToArraySizeMap() was
+     * called first to setup the source data.
+     */
+    protected void populateJsonMapArrays() {
+
+        Set<String> orderedKeys = arrayPathToArraySizeMap.keySet();
+        for (String key : orderedKeys) {
+            String[] steps = key.split("\\.");
+            if (steps.length > 0 && steps[0].equals("$")) {
+                steps = Arrays.copyOfRange(steps, 1, steps.length);
+            }
+            populateJsonMapArray(jsonMap, steps, new ArrayList<>(Arrays.asList("$")));
+        }
+    }
+
+    protected void populateJsonMapWithMockValues(ValidationSetModel validationSetModel) throws IOException, ClassNotFoundException {
+
+        NodeModel[] nodes = validationSetModel.getData();
+        for (NodeModel node : nodes) {
+
+            JsonBaseType jsonBaseType = convertNodeModelToJsonBaseType(node);
+            String[] savedPaths = jsonBaseType.getSavedPathsForNode();
+
+            for (String savedPath : savedPaths) {
+
+                JsonPathExecutor pathCheck = jsonBaseType.getCheckForPath(savedPath);
+                DeveloperMockTypeDecoder mockValueDecoder = null;
+                if (pathCheck instanceof DeveloperMockTypeDecoder) {
+                    mockValueDecoder = (DeveloperMockTypeDecoder) pathCheck;
+                }
+                String[] splitPath = savedPath.split("\\.");
+                Map rootNode = jsonMap;
+                walkAndPopulateJsonMap(rootNode, splitPath, new DeveloperMockValueLooper(mockValueDecoder));
+            }
+        }
+    }
+
+    protected void walkAndPopulateJsonMap(Map currentNode, String[] pathSteps, DeveloperMockValueLooper mockValueLooper) {
+
+        if (currentNode == null || pathSteps == null || mockValueLooper == null) {
```

Comment:
```suggestion
        if (currentNode == null || pathSteps == null || pathSteps.isEmpty() || mockValueLooper == null) {
```

Reply:
Had to use pathSteps.length == 0, but same effect. Thanks.